src/functions/transactor.ts

```diff
@@ -13,11 +14,32 @@ import {
 
 const callbacks: Record<string, any> = {};
 const DEBUG = true;
+
+export class TransactorError extends Error {
+  rawError: TRawTxError;
+
+  constructor({ message, rawError }: { message?: string; rawError: TRawTxError }) {
+    super();
+    this.message = message ?? 'Generic Error';
+    this.rawError = rawError;
+  }
+}
+
+export type TRawTxError = Error & {
+  data?: {
+    message?: string;
+  };
+};
+
 export type TTransactor = (
   tx: Deferrable<TransactionRequest> | Promise<TransactionResponse>,
   callback?: ((_param: any) => void) | undefined
 ) => Promise<Record<string, any> | TransactionResponse | undefined>;
 
+export type NotificationMessage = ArgsProps;

```

Comment:
Add ts docs to both :)

Reply:
if its internal (not exposed outside in docs use `@internal` in tsdocs)

Reply:
i create TSDoc explain it